libnd4j/include/ops/declarable/generic/convo/conv2d.cpp

```diff
@@ -63,18 +63,19 @@ CUSTOM_OP_IMPL(conv2d, 2, 1, false, 0, 9) {
     if (bias) 
         REQUIRE_TRUE(bias->rankOf() <= 2 && oC == bias->lengthOf(), 0, "CUSTOM CONV2D OP: wrong shape of array with biases, expected rank, length: <=2, %i, but got %i, %i instead !", oC, bias->rankOf(), bias->lengthOf());                
 
-// #ifdef HAVE_MKLDNN
-//     if (block.isUseMKLDNN() && MKLDNNStream<T>::isSupported()) {
-//         if (block.getMKLDNNStream() == nullptr) {
-//             block.setMKLDNNStream(new MKLDNNStream<T>("conv2d"));
-//         }
-//         ConvolutionUtils<T>::mkldnn_conv2d(*block.getMKLDNNStream(), {input, weights, bias}, output, {kH,kW,sH,sW,pH,pW,dH,dW,isSameMode,isNCHW});
-//     } else {
-// #endif
+#ifdef HAVE_MKLDNN
+    if (shape::strideDescendingCAscendingF(input->getShapeInfo()) && shape::strideDescendingCAscendingF(weights->getShapeInfo()) && shape::strideDescendingCAscendingF(bias->getShapeInfo()) && shape::strideDescendingCAscendingF(weights->getShapeInfo()) 
+        && shape::strideDescendingCAscendingF(output->getShapeInfo()) && block.isUseMKLDNN() && MKLDNNStream<T>::isSupported()) {
+        if (block.getMKLDNNStream() == nullptr) {
+            block.setMKLDNNStream(new MKLDNNStream<T>("conv2d"));
+        }
+        ConvolutionUtils<T>::mkldnn_conv2d(*block.getMKLDNNStream(), {input, weights, bias}, output, {kH,kW,sH,sW,pH,pW,dH,dW,isSameMode,isNCHW});
+    } else {
+#endif
```

Comment:
Please don't so that, MKL-DNN does support strides:
https://intel.github.io/mkl-dnn/understanding_memory_formats.html